### DISCUSSION TITLE
Sub menu's with hidden parent

### DISCUSSION THREAD

**Discussion by cnorthfield** [floating-ui/floating-ui#2779] [Q&A] [answered]
I'm trying to implement a list as shown in the [Dropdown Menu (including submenus example)](https://codesandbox.io/s/admiring-lamport-5wt3yg?file=/src/DropdownMenu.tsx). However, I'd like to know the best approach to hide the parent. I.e. when you select a list item (via a click or right arrow key event), the child list is shown, and the parent is hidden. When pressing the back arrow, it should show the parent list again.

Is this possible?

**Comment by cnorthfield** [comment]
I've also tried to change the list items based on state; however, when I switch the items, they lose focus.

Is there any guidance on how to show/hide different floating lists completely while being able to navigate back and forth with the mouse and keyboard?

**Reply by atomiks** [reply, marked as answer]
Sounds like you're looking for the non-cascading nested submenu system instead, like the YouTube settings menu/quality picker. It's possible but there's no demo for it; you can use `visibility: hidden` on the parent when a child opens and some absolute positioning to overlay the nested lists on top (with the parent container adapting based on the new list size etc.)

**Reply by cnorthfield** [reply]
Thanks @atomiks. I'll try out a few approaches and post back here what I found best for future reference.

**Reply by sladg** [reply]
Hey! working on same thing, have you had success implementing? Would you share? :) @cnorthfield

**Reply by cnorthfield** [reply]
Hey @sladg. Ultimately, I used separate components for each list.

I attached an event listener for the right arrow key and then when the right arrow key is pressed, the parent list is hidden and the child list is shown (controlled by state which holds the currently displayed list).

When the list changes, I reset the active index back to 0.
